PoseEstimation/poseEstimation_main.py

```diff
@@ -0,0 +1,182 @@
+
+    """
+    DeepLabCut algorithm
+    """
+    # --------------- Read danse parameters ---------------
+    Operations: str    = poseEstimation_params.params[defs.DoricFile.Attribute.Group.OPERATIONS]
+    filePath: str      = poseEstimation_params.paths[defs.Parameters.Path.FILEPATH]
+    datapath: str      = poseEstimation_params.paths[defs.Parameters.Path.H5PATH]
+    projectFolder: str = poseEstimation_params.params[poseEst_defs.Parameters.danse.PROJECT_FOLDER]
+    bodyPartNames      = poseEstimation_params.params[poseEst_defs.Parameters.danse.BODY_PART_NAMES].split(', ')
+    bodyPartColors     = poseEstimation_params.params[poseEst_defs.Parameters.danse.BODY_PART_COLORS].split(', ')
+    extractedFrames    = poseEstimation_params.params[poseEst_defs.Parameters.danse.EXTRACTED_FRAMES]
+    trainingCoordinates = {}
+    for bodyPart in bodyPartNames:
+        label = bodyPart + poseEst_defs.Parameters.danse.COORDINATES
+        trainingCoordinates[label] = poseEstimation_params.params[label]
+
+    # --------------- Create Project folder and config file ---------------
+    task: str        = "DLC" 
+    experimenter:str = "doric"
+
+    file_ = h5py.File(filePath, 'a')
+    attributes = utils.load_attributes(file_, datapath)
+    key = [key for key in attributes if poseEst_defs.Parameters.danse.VIDEO_FILEPATH in key]   
+    videoPath: str = attributes[key[0]]
+    # file_.close
+
+    path_config_file: str = deeplabcut.create_new_project(task, experimenter, [videoPath], projectFolder, copy_videos = False)
+    updateConfigFile(path_config_file, bodyPartNames)
+
+    # --------------- Create hdf file for labeled data ---------------
+    createlabeledDataHDF(path_config_file, extractedFrames, bodyPartNames, experimenter, poseEstimation_params, videoPath)
+
+    # --------------- Create a training dataset ---------------
+    deeplabcut.create_training_dataset(path_config_file)
+
+    # --------------- Start training ---------------
+    deeplabcut.train_network(path_config_file)
+
+    # --------------- Start evaluating ---------------
+    deeplabcut.evaluate_network(path_config_file, plotting=True)
+
+    # --------------- Start Analyzing videos ---------------
+    path_output = path_config_file.rsplit("\\", 1)[0]
+    deeplabcut.analyze_videos(path_config_file, [videoPath], destfolder = path_output)
+
+    # --------------- Saving data ---------------
+    saveCoordsToDoric(file_, datapath, path_output, extractedFrames, bodyPartNames, projectFolder, bodyPartColors, Operations, 
+                      trainingCoordinates, groupNames = poseEstimation_params.get_h5path_names())
+    
+    msg = ("************Process Completed*************************")
+    utils.print_to_intercept(msg)
+
+def preview(poseEstimation_params: poseEst_params.PoseEstimationParameters):
+    print("hello preview")
+
+def createlabeledDataHDF(path_config_file, extractedFrames, bodyPartNames, experimenter, poseEstimation_params, videoPath):
```

Comment:
@kapilseh have you seen this page? https://deeplabcut.github.io/DeepLabCut/docs/recipes/OtherData.html
I guess we can use built in function `deeplabcut.convertcsv2h5()`